google-cloud-spanner/src/main/java/com/google/cloud/spanner/AbstractReadContext.java

```diff
@@ -387,6 +387,8 @@ void initTransaction() {
   @GuardedBy("lock")
   protected boolean isClosed = false;
 
+  protected boolean readOnly;
```

Comment:
Can this be final?

Reply:
We'll not be able to change its value if it's final. We need to change its value in its subclasses to return whether it's read-only or not.